MVVM Module/Source/Feed Presentation/FeedRefreshViewModel.swift

```
@@ -0,0 +1,33 @@
+
+import Foundation
+import FeedFeature
+
+
+final class FeedRefreshViewModel {
+    
+    typealias Observer<T> = (T) -> Void
+    
+    private let feedLoader: FeedLoader
+    
+    init(feedLoader: FeedLoader) {
+        self.feedLoader = feedLoader
+    }
+
+    var onLoadingStateChange: Observer<Bool>?
+    var onFeedLoad: Observer<[FeedImage]>?
+    var onFeedError: Observer<String>?
```

`onFeedError: Observer<String> ` could be `onErrorStateChange: Observer<String?>`, so the ViewModel when to show/hide errors. It's often easier to manage state when all changes are centralized.